### PR TITLE
Add oidc authentication

### LINKED ISSUE #126: Provide alternative authorization with keycloak
Goal of the issue is to provide alternative authentication to the tool using keycloak module.

It should be done in a similar way as it was done for Record Manager, see https://github.com/kbss-cvut/record-manager/pull/5, https://github.com/kbss-cvut/record-manager-ui/pull/7.  Also create docker-compose set up for fta-fmea-ui in folder deploy/keycloak-auth as can be seen in https://github.com/kbss-cvut/record-manager-ui/tree/main/deploy/keycloak-auth.

Note:
- there exist already some docker-compose set up [here](https://github.com/kbss-cvut/19msmt-distribution/blob/master/docker-compose.yml) but it has more docker services defined than needed.
- you will need to switch database of fta-fmea-server to Graphdb (instead of RDF4j which was used until now). We did the same switch in Record Manager and we use now [this service](https://github.com/kbss-cvut/record-manager-ui/blob/129feca9f7ec901fe797c2977d07fdc9f1c34fd4/deploy/keycloak-auth/docker-compose.yml#L79). Switch from RDF4J to Graphdb should not require any change of code.
- keycloak authorization module was developed by us [here](https://github.com/kbss-cvut/keycloak-graphdb-user-replicator).  When an application user is configured using keycloak UI, the module automatically replicates some data into GraphDB database.
- at the end we would like to add fta-fmea-ui and fta-fmea-server to set up https://github.com/kbss-cvut/record-manager-ui/blob/main/deploy/keycloak-auth/docker-compose.yml, thus it is good if you follow the same convention (or propose change also to the Record Manager)

A/C:
- it is possible to configure fta-fma-ui to use keycloak authorization or internal authorization
- unrelated part of the UI is hidden/disabled when keycloak authorization is enabled 
- there is a docker-compose  set up in deploy/keycloak-auth that can be used run the tool. It should include database, fta-fmea-server and fta-fmea-ui. 
- graphdb is used in docker-compose set-up instead of rdf4j

### PR DESCRIPTION
**Pull request by LaChope**
Resolves #126. It mainly uses the same implementation as the [record-manager-ui](https://github.com/kbss-cvut/record-manager-ui).

I would not merge before the backend is also implemented to test it first.

### CODE REVIEW

File: deploy/keycloak-auth/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      DB_VENDOR: POSTGRES
+      DB_ADDR: auth-server-db
+      DB_DATABASE: keycloak
+      DB_USER: keycloak
+      DB_PASSWORD: keycloak
+      DB_SCHEMA: "public"
+      DB_SERVER_URL: "http://db-server:7200"
+      DB_SERVER_REPOSITORY_ID: "record-manager-app"
+      REPOSITORY_LANGUAGE: "en"
+      VOCABULARY_USER_TYPE: "http://onto.fel.cvut.cz/ontologies/record-manager/user"
```
Comment (blcham) [inline deploy/keycloak-auth/docker-compose.yml:114]: This might be an issue, we might need to align urls.

File: deploy/keycloak-auth/.env [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Prefix for name of all docker containers. By default it is set to "ff".
+RECORD_SET_NAME=ff-iauth-demo
```
Comment (blcham) [inline deploy/keycloak-auth/.env:2]:
```suggestion
RECORD_SET_NAME=ff-kauth-demo
```

File: src/utils/constants.tsx [hunk 3]
```diff
@@ -66,3 +71,8 @@ export const SVG_PAN_ZOOM_OPTIONS = {
 export const SELECTED_LANGUAGE_KEY = "default-fta-language";
 export const PRIMARY_LANGUAGE = "en";
 export const SECONDARY_LANGUAGE = "cs";
+
+export const ROLE = {
+  ADMIN: "Admin",
+  DOCTOR: "Regular User",
```
Comment (blcham) [inline src/utils/constants.tsx:77]: ???

**Review by blcham — CHANGES_REQUESTED** [review]
see my comments

**Comment by LaChope** [comment]
@blcham This PR was rebased

**Pull request merged**